warp10/src/main/java/io/warp10/script/functions/FORSTEP.java

```diff
@@ -106,9 +106,11 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     }
     
     long now = System.currentTimeMillis();
-    
+
+    long maxtime = this.maxtime > 0 ? this.maxtime : (long) stack.getAttribute(WarpScriptStack.ATTRIBUTE_LOOP_MAXDURATION);
+
     while (true) {
-      if (System.currentTimeMillis() - now > this.maxtime) {
+      if (System.currentTimeMillis() - now > maxtime) {
         throw new WarpScriptException(getName() + " executed for too long (> " + this.maxtime + " ms).");
```

Comment:
Error message should use maxtime, not this.maxtime

Reply:
Indeed. Fixed.